pkg/syncer/metadatasyncer.go

```diff
@@ -1048,23 +1047,46 @@ func csiUpdatePod(ctx context.Context, pod *v1.Pod, metadataSyncer *metadataSync
 					log.Errorf("failed to get volume id for volume name: %q with err=%v", pv.Name, err)
 					continue
 				}
-				updateSpec := &cnstypes.CnsVolumeMetadataUpdateSpec{
-					VolumeId: cnstypes.CnsVolumeId{
-						Id: volumeHandle,
-					},
-					Metadata: cnstypes.CnsVolumeMetadata{
-						ContainerCluster:      containerCluster,
-						ContainerClusterArray: []cnstypes.CnsContainerCluster{containerCluster},
-						EntityMetadata:        metadataList,
-					},
+			}
+		} else {
+			if volume.VsphereVolume != nil && metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) {
+				// Inline migrated volumes with no PVC
+				var podMetadata *cnstypes.CnsKubernetesEntityMetadata
+				if !deleteFlag {
+					podMetadata = cnsvsphere.GetCnsKubernetesEntityMetaData(pod.Name, nil, deleteFlag, string(cnstypes.CnsKubernetesEntityTypePOD), pod.Namespace, metadataSyncer.configInfo.Cfg.Global.ClusterID, nil)
```

Comment:
Have you verified whether entityReference is a mandatory field for Pod objects according to the CNS API?

Reply:
Yes, it is not a mandatory field for Pod objects